packages/babel-plugin-transform-typescript/test/fixtures/enum/outer-const-foldable/output.js

```diff
@@ -0,0 +1,15 @@
+// ts 5.0 feature
+
+const BaseValue = 10;
+const Prefix = "/data";
+var Values; // 12
+(function (Values) {
+  Values[Values["First"] = 10] = "First";
+  Values[Values["Second"] = 11] = "Second";
+  Values[Values["Third"] = 12] = "Third";
+})(Values || (Values = {}));
+var Routes; // "/data/invoices"
```

Comment:
An incorrect comment.

Reply:
![image](https://user-images.githubusercontent.com/30521560/215441131-f23be100-2966-4dad-a9bc-d0cc23b2f016.png)
The parser doesn't seem to be very good about this.
Comments are never considered trailing comments, but are treated as leading and inner comments.

Reply:
You can mark the inner comment `c` as a trailing comment of the enum member `c` here:

https://github.com/babel/babel/blob/41aac8b8037c566047a1f6792cc81c0e01613373/packages/babel-parser/src/parser/comments.ts#L165-L168

But for comment `a` and `b`, Babel parser does not differentiate the provided example with
```js
enum A {
  a,
  // a
  b,
  // b
  c
}
```
so I am afraid there aren't much we can do here.

Reply:
Thanks!
Yes, I found that too while doing this.
The babel parser does this even for plain objects.
This is expected, but not ideal, and I haven't figured out how to do it better, as it might mean a breaking change to the parser.